### PR TITLE
ARROW-13446: [Release] Fix verification on amazon linux

### PR DESCRIPTION
**Pull request by kszucs**
Quick fix for installing amazon linux packages.

### CODE REVIEW

File: dev/release/verify-yum.sh [hunk 1]
```diff
@@ -120,6 +124,10 @@ else
       -i"" \
       -e "s,/centos/,/centos-rc/,g" \
       /etc/yum.repos.d/Apache-Arrow.repo
+    sed \
+      -i"" \
```
Comment (kou) [inline dev/release/verify-yum.sh:128]:
```suggestion
```

File: dev/release/verify-yum.sh [hunk 2]
```diff
@@ -58,6 +57,8 @@ case "${distribution}-${distribution_version}" in
     have_gandiva=no
     have_python=no
     install_command="yum install -y"
+    distribution_prefix="amazon-linux"
+    amazon-linux-extras install epel -y
```
Comment (kou) [inline dev/release/verify-yum.sh:61]: Could you remove https://github.com/apache/arrow/blob/master/dev/release/verify-yum.sh#L92 ?

**Review by kou — APPROVED** [review]
+1

Oh, sorry. Thanks!

File: dev/release/verify-yum.sh (hunk 2 above)
Comment (kszucs) [reply dev/release/verify-yum.sh:61]: Otherwise it complains about missing epel.

Comment (kou) [reply dev/release/verify-yum.sh:61]: This addition is always executed with all `TYPE`s and https://github.com/apache/arrow/blob/master/dev/release/verify-yum.sh#L92 is executed only with `${TYPE} = "local"`. So https://github.com/apache/arrow/blob/master/dev/release/verify-yum.sh#L92 is needless by this addition.

**Comment by kou** [comment]
@github-actions crossbow submit amazon-linux-2-amd64

File: dev/release/verify-yum.sh (hunk 2 above)
Comment (kou) [reply dev/release/verify-yum.sh:61]: https://github.com/ursacomputing/crossbow/runs/3148979059?check_suite_focus=true#step:11:67 is by this addition.
https://github.com/ursacomputing/crossbow/runs/3148979059?check_suite_focus=true#step:11:193 is by https://github.com/apache/arrow/blob/master/dev/release/verify-yum.sh#L92 .

Comment (kszucs) [reply dev/release/verify-yum.sh:61]: If I remove that line I'm getting:

```
+ yum install -y https://apache.jfrog.io/artifactory/arrow/amazon-linux-rc/2/apache-arrow-release-latest.rpm
Loaded plugins: ovl, priorities
Examining /var/tmp/yum-root-BBDKLO/apache-arrow-release-latest.rpm: apache-arrow-release-5.0.0-1.amzn2.noarch
Marking /var/tmp/yum-root-BBDKLO/apache-arrow-release-latest.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package apache-arrow-release.noarch 0:5.0.0-1.amzn2 will be installed
--> Processing Dependency: epel-release for package: apache-arrow-release-5.0.0-1.amzn2.noarch
--> Processing Dependency: yum-utils for package: apache-arrow-release-5.0.0-1.amzn2.noarch
--> Running transaction check
---> Package apache-arrow-release.noarch 0:5.0.0-1.amzn2 will be installed
--> Processing Dependency: epel-release for package: apache-arrow-release-5.0.0-1.amzn2.noarch
---> Package yum-utils.noarch 0:1.1.31-46.amzn2.0.1 will be installed
--> Processing Dependency: python-kitchen for package: yum-utils-1.1.31-46.amzn2.0.1.noarch
--> Processing Dependency: libxml2-python for package: yum-utils-1.1.31-46.amzn2.0.1.noarch
--> Running transaction check
---> Package apache-arrow-release.noarch 0:5.0.0-1.amzn2 will be installed
--> Processing Dependency: epel-release for package: apache-arrow-release-5.0.0-1.amzn2.noarch
---> Package libxml2-python.x86_64 0:2.9.1-6.amzn2.5.4 will be installed
---> Package python-kitchen.noarch 0:1.1.1-5.amzn2 will be installed
--> Processing Dependency: python-chardet for package: python-kitchen-1.1.1-5.amzn2.noarch
--> Running transaction check
---> Package apache-arrow-release.noarch 0:5.0.0-1.amzn2 will be installed
--> Processing Dependency: epel-release for package: apache-arrow-release-5.0.0-1.amzn2.noarch
---> Package python-chardet.noarch 0:2.2.1-1.amzn2 will be installed
--> Finished Dependency Resolution
Error: Package: apache-arrow-release-5.0.0-1.amzn2.noarch (/apache-arrow-release-latest)
           Requires: epel-release
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
 ```

Comment (kszucs) [reply dev/release/verify-yum.sh:61]: Seems like @nealrichardson has merged it :)
@kou please create a follow-up if we need to fix this.

Comment (kou) [reply dev/release/verify-yum.sh:61]: OK.
The error message shows that we need both of them. I was wrong. :-)

**Pull request closed without merging**